storio-common/src/test/java/com/pushtorefresh/storio/operations/internal/OnSubscribeExecuteAsBlockingSingleTest.java

```diff
@@ -0,0 +1,68 @@
+package com.pushtorefresh.storio.operations.internal;
+
+import com.pushtorefresh.storio.StorIOException;
+import com.pushtorefresh.storio.operations.PreparedOperation;
+
+import org.junit.Test;
+
+import rx.Single;
+import rx.observers.TestSubscriber;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+public class OnSubscribeExecuteAsBlockingSingleTest {
+
+    @SuppressWarnings("CheckResult")
+    @Test
+    public void shouldExecuteAsBlockingAfterSubscription() {
+        //noinspection unchecked
+        final PreparedOperation<Object> preparedOperation = mock(PreparedOperation.class);
+
+        TestSubscriber<Object> testSubscriber = new TestSubscriber<Object>();
+
+        verifyZeroInteractions(preparedOperation);
+
+        Single<Object> single = Single.create(OnSubscribeExecuteAsBlockingSingle.newInstance(preparedOperation));
+
+        verifyZeroInteractions(preparedOperation);
+
+        single.subscribe(testSubscriber);
+
+        testSubscriber.assertNoErrors();
+        testSubscriber.assertCompleted();
+
+        verify(preparedOperation).executeAsBlocking();
+        verify(preparedOperation, never()).asRxObservable();
+        verify(preparedOperation, never()).asRxSingle();
+    }
+
+    @SuppressWarnings("CheckResult")
+    @Test
+    public void shouldCallOnErrorIfExceptionOccurred() {
+        //noinspection unchecked
+        final PreparedOperation<Object>  preparedOperation = mock(PreparedOperation.class);
```

Comment:
Oh man, please review this PR so we could merge it and include to next
release

On Fri, 8 Jul 2016, 02:04 Rustam, <notifications@github.com> wrote:

> In
> storio-common/src/test/java/com/pushtorefresh/storio/operations/internal/OnSubscribeExecuteAsBlockingSingleTest.java
> <https://github.com/pushtorefresh/storio/pull/601#discussion_r70003024>:
>
> > +
> > +        single.subscribe(testSubscriber);
> > +
> > +        testSubscriber.assertNoErrors();
> > +        testSubscriber.assertCompleted();
> > +
> > +        verify(preparedOperation).executeAsBlocking();
> > +        verify(preparedOperation, never()).asRxObservable();
> > +        verify(preparedOperation, never()).asRxSingle();
> > +    }
> > +
> > +    @SuppressWarnings("CheckResult")
> > +    @Test
> > +    public void shouldCallOnErrorIfExceptionOccurred() {
> > +        //noinspection unchecked
> > +        final PreparedOperation<Object>  preparedOperation = mock(PreparedOperation.class);
>
> empty space =)
>
> —
> You are receiving this because you were mentioned.
>
> Reply to this email directly, view it on GitHub
> <https://github.com/pushtorefresh/storio/pull/601/files/220e648e1805c9bfbfd9d69dca781562b677225b#r70003024>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA7B3Bavd02WHkY62nqdS7pxjUen_tcKks5qTYYNgaJpZM4HK7aS>
> .
>